### ISSUE TITLE
Graph used for SMT formulation does not include all iBGP adjacencies

### ISSUE THREAD

**Issue by agember** [batfish/batfish#2504]
An exception occurs when asking an smt-reachability question due to an unknown iBGP neighbor. All expected layer-3 and BGP adjacencies are reported when asking an `edges` question, but the same BGP adjacencies are not included in the graph used by the SMT code. The exception trace is below. I can provide additional output and configs that demonstrate this error on a private channel.

```
org.batfish.common.BatfishException: Failed to answer question
    at org.batfish.main.Batfish.answer(Batfish.java:624)
    at org.batfish.main.Batfish.run(Batfish.java:3081)
    at org.batfish.main.Driver.lambda$runBatfish$3(Driver.java:580)
    at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NullPointerException
    at org.batfish.symbolic.smt.EncoderSlice.otherSliceHasEdge(EncoderSlice.java:1670)
    at org.batfish.symbolic.smt.EncoderSlice.addDataForwardingConstraints(EncoderSlice.java:1731)
    at org.batfish.symbolic.smt.EncoderSlice.computeEncoding(EncoderSlice.java:2327)
    at org.batfish.symbolic.smt.Encoder.computeEncoding(Encoder.java:842)
    at org.batfish.symbolic.smt.PropertyChecker.lambda$checkProperty$2(PropertyChecker.java:410)
    at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
    at java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1359)
    at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
    at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:498)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:485)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
    at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
    at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:449)
    at org.batfish.symbolic.smt.PropertyChecker.checkProperty(PropertyChecker.java:393)
    at org.batfish.symbolic.smt.PropertyChecker.checkReachability(PropertyChecker.java:557)
    at org.batfish.main.Batfish.smtReachability(Batfish.java:3776)
    at org.batfish.question.SmtReachabilityQuestionPlugin$ReachabilityAnswerer.answer(SmtReachabilityQuestionPlugin.java:33)
    at org.batfish.question.jsonpath.JsonPathQuestionPlugin$JsonPathAnswerer.answer(JsonPathQuestionPlugin.java:164)
    at org.batfish.question.jsonpath.JsonPathQuestionPlugin$JsonPathAnswerer.answer(JsonPathQuestionPlugin.java:135)
    at org.batfish.main.Batfish.answer(Batfish.java:621)
    ... 3 more
```

**Comment by ratulm** [comment]
hey, @agember - is this a duplicate of #2503?

hey, @rabeckett - you know what is happening here?

Cc: @anothermattbrown

**Comment by agember** [comment]
@ratulm Sorry, GitHub wasn't showing the issue, so I created it again, but apparently it duplicated. I closed #2503.

**Comment by dhalperi** [comment]
(@ratulm injecting, with the current level of support for the SMT code, we're likely to delete it from mainline `master` repository. Either we find a capable owner for it or we delete it so folks don't think we're still supporting it...)

**Comment by agember** [comment]
I'm happy to dig into the SMT code to debug this issue if someone can provide some pointers as to where to start. Mostly I was surprised by the disagreement between the two views of adjacencies; I assumed the SMT formulation's view of connectivity would be based on the same view of connectivity used elsewhere in the code.

**Comment by rabeckett** [comment]
I'm not sure what is happening just from the trace, but the code that initializes ibgp adjacencies for the Graph class is here:

https://github.com/batfish/batfish/blob/master/projects/batfish/src/main/java/org/batfish/symbolic/Graph.java#L573

**Comment by progwriter** [comment]
Fixed in #2562, an extension of #2550
